django/db/backends/base/base.py

```diff
@@ -510,12 +515,25 @@ def is_usable(self):
         raise NotImplementedError(
             "subclasses of BaseDatabaseWrapper may require an is_usable() method")
 
+    def close_if_health_check_failed(self):
+        if not (self.connection and
+                self.health_checks_on and
+                not self.health_check_done):
+            return
+
+        if not self.is_usable():
+            self.close()
```

Comment:
`.close()` already handles backend specific `connection.close()` errors and clears `self.connection`. This call here is not different from the call to `.close()` from inside of `close_if_unusable_or_obsolete()`. So I think we should be good here. What am I missing?

Reply:
I was considering the case where close might fail and as a result does not set `self.connection` to `None` because it returns before that (probably via raise). But if `close_if_unusable_or_obsolete()` doesn't handle it any differently I guess we are fine -- we can add it later if needed.